common/src/main/java/edu/snu/onyx/common/ir/vertex/InitializedSourceVertex.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2017 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.onyx.common.ir.vertex;
+
+import edu.snu.onyx.common.ir.Reader;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Source vertex with initial data as object.
+ * @param <T> type of initial data.
+ */
+public final class InitializedSourceVertex<T> extends SourceVertex<T> {
+  private final Iterable<T> initializedSourceData;
+
+  /**
+   * Constructor.
+   * @param initializedSourceData the initial data object.
+   */
+  public InitializedSourceVertex(final Iterable<T> initializedSourceData) {
+    this.initializedSourceData = initializedSourceData;
+  }
+
+  @Override
+  public InitializedSourceVertex<T> getClone() {
+    final InitializedSourceVertex<T> that = new InitializedSourceVertex<>(this.initializedSourceData);
+    this.copyExecutionPropertiesTo(that);
+    return that;
+  }
+
+  @Override
+  public List<Reader<T>> getReaders(final int desiredNumOfSplits) throws Exception {
+    final List<Reader<T>> readers = new ArrayList<>();
+    final long sliceSize = initializedSourceData.spliterator().getExactSizeIfKnown() / desiredNumOfSplits;
+    final Iterator<T> iterator = initializedSourceData.iterator();
+
+    for (int i = 0; i < desiredNumOfSplits; i++) {
```

Comment:
I've put it this way, since there isn't really a `Source` for this type of source vertex. The source data is the initialized data that is provided to the vertex as a Java object. Although it seems more simple in the `BoundedSourceVertex` example, it is because the Beam's implementations are hidden underneath (its `split()` method is over 50 lines of code). We will have to implement the same thing in a separate class, which I think makes it unnecessarily more complex. What do you think?

Reply:
I understood. Let's move on as it is.
Thanks for the explanation.